st2common/st2common/models/db/auth.py

```diff
@@ -49,4 +53,32 @@ class TokenDB(stormbase.StormFoundationDB):
                             help_text='Arbitrary metadata associated with this token')
 
 
-MODELS = [UserDB, TokenDB]
+class ApiKeyDB(stormbase.StormFoundationDB, stormbase.UIDFieldMixin):
+    """
+    """
+    RESOURCE_TYPE = ResourceType.API_KEY
+    UID_FIELDS = ['id']
+
+    user = me.StringField(required=True)
+    key_hash = me.StringField(required=True, unique=True)
+    metadata = me.DictField(required=False,
```

Comment:
I would prefer `description` over metadata but I just noticed that token also used metadata so that's fine.

On top of that, as noted above I think created_at and enabled attribute would also come handy here.